src/components/Dialog/DialogFooter/DialogFooter.tsx

```diff
@@ -55,128 +53,103 @@ function getButtonView(preset: ButtonPreset): ButtonView {
+        onClickButtonCancel,
+        onClickButtonApply,
+        loading,
+        textButtonCancel,
+        textButtonApply,
+        propsButtonCancel,
+        propsButtonApply,
+        preset = 'default',
+        children,
+        errorText,
+        showError = false,
+        renderButtons,
+        className,
+    } = props;
+
+    const {initialFocusRef, initialFocusAction, onTooltipEscapeKeyDown} =
+        React.useContext(DialogPrivateContext);
+
+    const errorTooltipRef = React.useRef<HTMLButtonElement>(null);
+    const apllyBtnRef = useForkRef(
+        errorTooltipRef,
+        initialFocusAction === 'apply' ? initialFocusRef : null,
+    );
+    const cancelBtnRef = useForkRef(initialFocusAction === 'cancel' ? initialFocusRef : null);
+
+    const buttonCancel = (
+        <div className={b('button', {action: 'cancel'})}>
+            <Button
+                ref={cancelBtnRef}
+                view={textButtonApply ? 'flat' : 'normal'}
+                size="l"
+                width="max"
+                onClick={onClickButtonCancel}
+                disabled={loading}
+                {...propsButtonCancel}
+            >
+                {textButtonCancel}
+            </Button>
+        </div>
+    );
+
+    const handleOpenChange = React.useCallback<NonNullable<UseFloatingOptions['onOpenChange']>>(
+        (isOpen, event, reason) => {
+            if (!isOpen && event && reason === 'escape-key') {
+                onTooltipEscapeKeyDown?.(event as KeyboardEvent);
+            }
+        },
+        [onTooltipEscapeKeyDown],
+    );
+
+    const buttonApply = (
+        <div className={b('button', {action: 'apply'})}>
+            <Button
+                ref={apllyBtnRef}
+                type="submit"
+                view={getButtonView(preset)}
+                size="l"
+                width="max"
+                onClick={onClickButtonApply}
+                loading={loading}
+                className={b('button-apply', {preset})}
+                {...propsButtonApply}
+            >
+                {textButtonApply}
+            </Button>
+            {errorText && (
+                <Popup
+                    open={showError}
+                    onOpenChange={handleOpenChange}
+                    anchorRef={errorTooltipRef}
+                    placement={['bottom', 'top']}
+                    disablePortal
+                    hasArrow
                 >
-                    {textButtonApply}
-                </Button>
-                {errorText && (
-                    <Popup
-                        open={showError}
-                        anchorRef={this.errorTooltipRef}
-                        placement={['bottom', 'top']}
-                        disablePortal
-                        hasArrow
-                    >
-                        <div className={b('error')}>{errorText}</div>
-                    </Popup>
+                    <div className={b('error')}>{errorText}</div>
```

Comment:
Is it really should be part of base component? May be we should consider some render prop?